packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js

```diff
@@ -50,13 +50,43 @@ export default async (req, res) => {
     res.sendStatus(200)
   }
 
+  async function handlePaymentCollection(event, req, res, id, paymentIntentId) {
+    const manager = req.scope.resolve("manager")
+    const paymentCollectionService = req.scope.resolve(
+      "paymentCollectionService"
+    )
+
+    const paycol = await paymentCollectionService
+      .retrieve(id, { relations: ["payments"] })
+      .catch(() => undefined)
+
+    if (paycol?.payments?.length) {
+      if (event.type === "payment_intent.succeeded") {
+        const payment = paycol.payments.find(
+          (pay) => pay.data.id === paymentIntentId
+        )
+        if (payment && !payment.captured_at) {
+          await manager.transaction(async (manager) => {
+            await paymentCollectionService
+              .withTransaction(manager)
+              .capture(payment.id)
+          })
+        }
+
+        res.sendStatus(200)
+        return
+      }
+    }
```

Comment:
thought: how will we handle this case:
1. Customer clicks pay and immediately closes window
2. Payment provider authorization succeeds but `/store/carts/:id/complete` is not called
4. Webhook comes in notifying that the payment was processed
5. The webhook should now ensure that the payment is created in Medusa and that an order is created from the cart

Reply:
Just to clarify, we are not handling carts payment with PaymentCollection yet. But the flow you just described will be handled like the current Cart payment is.
When the webhook is notified, we get the payment collection, if the type is "`CART`" we just complete the flow.

Reply:
Got it - just wanted to make sure; I guess a similar thing would have to be done for OrderEdit flow though? I.e. when customer pays for the OrderEdit it should be possible to register the payment as authorized asynchronously.